voting_metadata.go

```diff
@@ -0,0 +1,49 @@
+package pbft
+
+type VotingMetadata interface {
+	// QuorumSize returns PBFT message count needed to perform a single PBFT state transition
+	QuorumSize() uint64
```

Comment:
I think these methods belong to the `ValidatorSet` object. These are fields of the validator set.

Reply:
On the other hand, think that `votingPower` map should be part of `ValidatorSet` object (as well) and that `VotingMetadata` should receive it from it.

Reply:
I aggree with  @ferranbt's point. It'd be great to move VotingMetadata to ValidatorSet interface. Initially i was thinking about the same but  tried not to change ValidatorSet interface. 
If we agreed about changing it we may simplify an interface to 1-2 functions. 
Like `HasQuorum(senders) bool`
`HasMaxFaultyVotingPower(senders) bool`
We get ValidatorSet for every block and  it knows when we reached quorum.

Reply:
If I understand your proposal well, don't like it just because I really think that pbft consensus library needs to offer (at least default) implementation for functions contained in `VotingMetadata` interface. Because otherwise library won't be encapsulated and self contained (if we were up to allow consumer of library to implement it arbitrarily).